libs/analytics/src/gtm/initGtm.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * GTM initialization module
+ * Provides type-safe initialization of Google Tag Manager
+ * Ensures single initialization of GTM instance
+ */
+
+import { CowAnalyticsGtm } from './CowAnalyticsGtm'
+
+import { CowAnalytics } from '../CowAnalytics'
+
+declare global {
+  interface Window {
+    dataLayer: unknown[]
+  }
+}
+
+const DEFAULT_GTM_ID = 'GTM-TBX4BV5M'
+
+// Module-level singleton state
+const analytics = {
+  instance: null as CowAnalytics | null,
+  isInitializing: false,
+  gtmId: DEFAULT_GTM_ID,
+}
+
+/**
+ * Initialize GTM and return a CowAnalytics instance
+ * This function ensures GTM is initialized only once
+ * @param gtmId - Optional GTM container ID
+ * @returns CowAnalytics instance backed by GTM
+ * @throws Error if attempting to initialize with different GTM ID
+ */
+export function initGtm(gtmId: string = DEFAULT_GTM_ID): CowAnalytics {
+  // Early return for SSR
+  if (typeof window === 'undefined') {
+    return new CowAnalyticsGtm()
+  }
+
+  // Return existing instance if already initialized
+  if (analytics.instance) {
+    if (gtmId !== analytics.gtmId) {
+      throw new Error('GTM already initialized with different ID')
+    }
+    return analytics.instance
+  }
+
+  // Prevent concurrent initializations
+  if (analytics.isInitializing) {
+    throw new Error('GTM initialization already in progress')
```

Comment:
This error should never be thrown since the code bellow is synchronous.
So we don't actually need `isInitializing` property.

Reply:
Addressed